src/engine/renderer/Material.cpp

```diff
@@ -214,45 +214,45 @@ void UpdateSurfaceDataGeneric3D( uint32_t* materials, Material& material, drawSu
 	}
 	drawSurf->initialized[stage] = true;
 
-	gl_genericShaderMaterial->BindProgram( material.deformIndex );
+	gl_generic3DShaderMaterial->BindProgram( material.deformIndex );
 
-	gl_genericShaderMaterial->SetUniform_ModelMatrix( backEnd.orientation.transformMatrix );
-	gl_genericShaderMaterial->SetUniform_ModelViewProjectionMatrix( glState.modelViewProjectionMatrix[glState.stackIndex] );
+	gl_generic3DShaderMaterial->SetUniform_ModelMatrix( backEnd.orientation.transformMatrix );
+	gl_generic3DShaderMaterial->SetUniform_ModelViewProjectionMatrix( glState.modelViewProjectionMatrix[glState.stackIndex] );
 
 	// u_AlphaThreshold
-	gl_genericShaderMaterial->SetUniform_AlphaTest( pStage->stateBits );
+	gl_generic3DShaderMaterial->SetUniform_AlphaTest( pStage->stateBits );
 
-	// u_InverseLightFactor
-	float inverseLightFactor = pStage->cancelOverBright ? tr.mapInverseLightFactor : 1.0f;
-	gl_genericShaderMaterial->SetUniform_InverseLightFactor( inverseLightFactor );
+	// u_LightFactor
+	float lightFactor = ( pStage->type == stageType_t::ST_STYLELIGHTMAP || pStage->type == stageType_t::ST_STYLECOLORMAP || ( tess.bspSurface && pStage->rgbGen == colorGen_t::CGEN_VERTEX ) ) ? tr.mapLightFactor : 1.0f;
```

Comment:
Use `drawSurf->bspSurface` here, `tess.bspSurface` won't be valid in this functions.